docs/design/queue/hierarchy-queue.md

```diff
@@ -0,0 +1,139 @@
+
+### Features of hierarchy queue
+
+1. `root` is a top-level parent queue which represents the cluster itself. All other queues are direct or indirect child queues of `root`.
+2. Initially, we only have one queue called `default` which belongs to `root`.
+3. `ParentQueue` is optional. If its value is null when you create a queue, it defaults to `root` as its parent queue.
+4. `ChildQueues` is optional. If its value is null when you create a queue, the queue will be a leaf queue.
+
+### QueueController
+
+1. Watching `PodGroup/Job` for status; The total number of `PodGroup`/`Job` owned by child queues is the number of `PodGroup`/`Job` owned by the parent queue.
+2. If queue was deleted, also delete all related `PodGroup/Job` and child queues of this queue.
+3. If queue was deleted and its parent queue will become a leaf queue, its parent queue's `ChildQueues` will change and its `MinUserLimitPercent`/`UserLimitFactor` will be valid.
+4. If a parent queue is closed, its child queues will be closed too.
+
+### Admission Controller
+
+The admission controller will check `PodGroup`/`Job` 's queue when creation:
+
+1. If the queue is parent queue, the creation will be rejected.
+2. If no queue is specified, it will be scheduled to the `root.default` queue.
+
+The admission controller will check other related queues when the queue creates:
+
+1. If this queue's `ParentQueue` is a leaf queue, and it has some running jobs, the creation will be rejected.
+2. If this queue meets the creation requirements, and its `ParentQueue` is a leaf queue. Its parent queue's `ChildQueues` will change and its `MinJobLimitPercent`/`JobLimitFactor` will be set to null.
+
+### Cli
+
+1. Because child queue's resource allocation depends on its parent queue's. We need to understand the hierarchy of each queue so that the administrator can clearly assign weights to queues.
+2. We need to know who leaf queue is, because just leaf queue can accept applications.
+
+So when we use `vqueues` to list all queues of the cluster, we can just use a display similar to `tree` command. Look at the following example.
+
+```shell
+$ vqueues
+Name             Weight  State  Inqueue  Pending  Running  Unknown
+root                  1   Open        6        0        6        0
+|--default            5   Open        3        0        3        0
+|--dev                5   Open        3        0        3        0
+|  |--test1           1   Open        1        0        1        0
+|  |--test2           2   Open        2        0        2        0 
+```
+
+## How to create?
+
+You can create hierarchy queues one by one with yaml. Or you can use `annotations` to define hierarchy queue. Here is an example for creating a leaf queue and its parent queues.
+
+```yaml
+apiVersion: scheduling.volcano.sh/v1beta1
+kind: Queue
+metadata:
+  name: test1
+  annotations:
+    "volcano.sh/hierarchy-queue": "root/dev/test1"
```

Comment:
it's better to use spec instead of annotation.

Reply:
Do you mean replace `ParentQueue` and `ChildQueues` with `spec.hierarchy-queue` ？And also use `spec.hierarchy-queue-weight` instead of annotation?